container/openejb-core/src/main/java/org/apache/openejb/assembler/classic/Assembler.java

```diff
@@ -1049,6 +1058,19 @@ private AppContext createApplication(final AppInfo appInfo, ClassLoader classLoa
         }
     }
 
+    private void ensureAppNamingContextIsReadOnly(final List<BeanContext> allDeployments) {
+        for(BeanContext beanContext : allDeployments) {
+            Context ctx = beanContext.getJndiContext();
+            if(ctx instanceof IvmContext) {
+               ((IvmContext) ctx).setReadOnly(true);
+            } else if(ctx instanceof ContextHandler) {
+            ((ContextHandler)ctx).setReadOnly();
```

Comment:
side note on casting: on the list we got a discussion saying we prefer the ${type}.class.cast(instance) style instead of (($type) instance). Not a blocker for me but just mentionning it if you see that style in the code elsewhere.